AdvancedTeleport-Bukkit/src/main/java/io/github/niestrat99/advancedteleport/commands/core/PurgeCommand.java

```diff
@@ -0,0 +1,146 @@
+
+                        @Override
+                        public void onFail() {
+                            CustomMessages.sendMessage(sender, "Error.purgeWarpsFail");
+                        }
+                    });
+                    break;
+            }
+
+        } else if (args[1].equalsIgnoreCase("player")) {
+            Bukkit.getScheduler().runTaskAsynchronously(CoreClass.getInstance(), () -> {
+                OfflinePlayer player = Bukkit.getOfflinePlayer(args[2]);
+                if (player.getName() == null) {
+                    CustomMessages.sendMessage(sender, "Error.noSuchPlayer");
+                    return;
+                }
+                switch (args[0].toLowerCase()) {
+                    case "homes":
+                        HomeSQLManager.get().purgeHomes(player.getUniqueId(), new SQLManager.SQLCallback<Void>() {
+                            @Override
+                            public void onSuccess(Void data) {
+                                CustomMessages.sendMessage(sender, "Info.purgeHomesCreator", "{player}", args[2]);
+                            }
+
+                            @Override
+                            public void onFail() {
+                                CustomMessages.sendMessage(sender, "Error.purgeHomesFail");
+                            }
+                        });
+                        break;
+                    case "warps":
+                        WarpSQLManager.get().purgeWarps(player.getUniqueId(), new SQLManager.SQLCallback<Void>() {
+                            @Override
+                            public void onSuccess(Void data) {
+                                CustomMessages.sendMessage(sender, "Info.purgeWarpsCreator", "{player}", args[2]);
+                            }
+
+                            @Override
+                            public void onFail() {
+                                CustomMessages.sendMessage(sender, "Error.purgeWarpsFail");
+                            }
+                        });
+                        break;
+                }
+            });
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String s, @NotNull String[] args) {
+        List<String> results = new ArrayList<>();
+
+        if (args.length == 1) {
+            StringUtil.copyPartialMatches(args[0], Arrays.asList("homes", "warps"), results);
+
+        } else if (args.length == 2 ) {
+            StringUtil.copyPartialMatches(args[1], Arrays.asList("player", "world"), results);
+
+        } else if (args.length == 3 ) {
+            if (args[1].equalsIgnoreCase("player")) {
+                List<String> players = new ArrayList<>();
+                Bukkit.getOnlinePlayers().forEach(player -> players.add(player.getName()));
+                StringUtil.copyPartialMatches(args[2], players , results);
```

Comment:
May need to reformat the file, do this by doing Code > Reformat code in IntelliJ